rust-examples/nocore-hello-world.rs

```diff
@@ -45,6 +45,9 @@ mod wasm {
     }
 }
 
+// This function will be run at Module start
+// e.g when using the interpreter
+#[wasm_start]
```

Comment:
the `#[start]` attribute on fns doesn't require a signature that matches the wasm `start` function — while `#[main]` does; it has to be void(void) — so this allows to have your "main" rust fn and main wasm fn, possibly different; Same deal as before it was mostly to make testing easier :) (and was already used in an example).
`#[main]` is already made the wasm start automatically, and this probably wouldn't replace #[start] but supplement it, yes, absolutely!